python/paddle/fluid/layers/tensor.py

```diff
@@ -569,9 +569,15 @@ def assign(input, output=None):
           result3 = paddle.assign(np.array([[2.5, 2.5], [2.5, 2.5], [2.5, 2.5]], dtype='float32')) # result3 = [[2.5, 2.5], [2.5, 2.5], [2.5, 2.5]]
     """
     helper = LayerHelper('assign', **locals())
-    check_type(input, 'input', (Variable, numpy.ndarray), 'assign')
+    check_type(input, 'input', (Variable, numpy.ndarray, list, tuple, float,
+                                int, bool), 'assign')
     is_inplace = True if output is not None else False
 
+    if numpy.isscalar(input) and not isinstance(input, str):
```

Comment:
It's not necessary  to check whether `input` is str because it has been checked in line 572.
`if numpy.isscalar(input) and not isinstance(input, str)`  -> `if numpy.isscalar(input)`.

Reply:
No, we cannot. Because `numpy.isscalar(input)` returns `True` when `input` is a string.